api/model.go

```diff
@@ -1072,6 +1078,28 @@ func toSnapshotCollection(ss map[string]*types.Snapshot) *client.GenericCollecti
 	return &client.GenericCollection{Data: data, Collection: client.Collection{ResourceType: "snapshot"}}
 }
 
+func toBackupTargetResource(bt *longhorn.BackupTarget) *BackupTarget {
+	if bt == nil {
+		logrus.Warnf("weird: nil backupTarget")
+		return nil
+	}
+	res := &BackupTarget{
+		Resource: client.Resource{
+			Id:    bt.Name,
+			Type:  "backupTarget",
+			Links: map[string]string{},
+		},
+		BackupTarget: engineapi.BackupTarget{
+			BackupTargetURL:  bt.Spec.BackupTargetURL,
+			CredentialSecret: bt.Spec.CredentialSecret,
+			PollInterval:     bt.Spec.PollInterval.Duration.String(),
+			Available:        bt.Status.Available,
+			Message:          bt.Status.Conditions[types.BackupTargetConditionTypeUnavailable].Message,
```

Comment:
If the condition unavailable, have a default message to avoid null pointer exception.

Reply:
I think NPE won't happens https://play.golang.org/p/0gO7qqleBxo